go/tasks/plugins/array/k8s/subtask.go

```diff
@@ -186,6 +186,17 @@ func launchSubtask(ctx context.Context, stCtx SubTaskExecutionContext, cfg *Conf
 		Value: strconv.Itoa(stCtx.originalIndex),
 	})
 
+	for sidecarIndex, container := range pod.Spec.Containers {
+		if container.Name == config.GetK8sPluginConfig().CoPilot.NamePrefix+flytek8s.Sidecar {
+			for i, arg := range pod.Spec.Containers[sidecarIndex].Args {
```

Comment:
cc @hamersaw should we pass env `FlyteK8sArrayIndex` to copilot, and construct final output prefix in the copilot?

Reply:
I don't have strong feelings here. How are we passing the array index to the inputs downloader? Because in flytekit we pass the input data ref and a subtask index, IIUC it reads the full list of inputs and only uses the value at the subtask index. We need to do the same thing here right?

Reply:
we pass array index to [primary container](https://github.com/flyteorg/flyteplugins/blob/d20137836713ba83acdd029cb1f83eb28b82ceb4/go/tasks/plugins/array/k8s/subtask.go#L183-L188), and raw container task will read the value at subtask index. here is an example. https://github.com/flyteorg/flytekit/pull/1547.

The problem is that in the regular map task, we construct the final output prefix in the flytekit (output_prefix + array index.), but the raw container doesn't know the output prefix, it write to a local share dir instead. uploader will read the data in the share dir and upload to s3.